server/datastore/mysql/microsoft_mdm.go

```diff
@@ -565,7 +570,20 @@ AND ` + whereKeyAvailable + `
 AND (
     (` + whereEncrypted + ` AND NOT ` + whereHostDisksUpdated + `)
     OR (NOT ` + whereEncrypted + ` AND ` + whereHostDisksUpdated + ` AND ` + withinGracePeriod + `)
-)`
+) 
+AND ` + whereBitLockerPINSet
+
+	case fleet.DiskEncryptionActionRequired:
+		// Action required means we _would_ be in verified / verifying,
+		// but we require a PIN to be set and it's not.
+		return whereNotServer + `
+AND NOT ` + whereClientError + `
+AND ` + whereKeyAvailable + `
+AND (
+	` + whereEncrypted + ` 
+	OR (NOT ` + whereEncrypted + ` AND ` + whereHostDisksUpdated + ` AND ` + withinGracePeriod + `)
+)
+AND NOT ` + whereBitLockerPINSet
```

Comment:
Add the new "action required" state, which is basically the disjunction of the "verifying" and "verified" states, with the additional "PIN not set" check.